patroni/ha.py

```diff
@@ -319,30 +339,38 @@ def should_run_scheduled_action(self, action_name, scheduled_at, cleanup_fn):
     def process_manual_failover_from_leader(self):
         failover = self.cluster.failover
 
+        if failover.scheduled_at and self.is_paused():
+            return
+
         if (failover.scheduled_at and not
             self.should_run_scheduled_action("failover", failover.scheduled_at, lambda:
```

Comment:
I'd recommend to add a check for is_paused inside the self.should_run_scheduled_action. It will be triggered then for both scheduled restart and failover (no need do add the explicit checks in the scheduled restart), and will also run the cleanup callback, which will not be run in your case.